specification/azurearcdata/resource-manager/Microsoft.AzureArcData/preview/2022-03-01-preview/azurearcdata.json

```diff
@@ -0,0 +1,1159 @@
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AzureArcData/dataControllers/{dataControllerName}/activeDirectoryConnectors/{activeDirectoryConnectorName}": {
+      "put": {
+        "tags": [
+          "ActiveDirectoryConnectors"
+        ],
+        "operationId": "ActiveDirectoryConnectors_Create",
+        "description": "Creates or replaces an Active Directory connector resource.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "name": "activeDirectoryConnectorResource",
+            "description": "desc",
+            "required": true,
+            "in": "body",
+            "schema": {
+              "$ref": "./activeDirectoryConnectors.json#/definitions/ActiveDirectoryConnectorResource"
+            }
+          },
+          {
+            "$ref": "#/parameters/dataControllerName"
+          },
+          {
+            "$ref": "#/parameters/activeDirectoryConnectorName"
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "./activeDirectoryConnectors.json#/definitions/ActiveDirectoryConnectorResource"
+            }
+          },
+          "201": {
+            "description": "Created",
+            "schema": {
+              "$ref": "./activeDirectoryConnectors.json#/definitions/ActiveDirectoryConnectorResource"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***",
+            "schema": {
+              "$ref": "./common.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "azure-async-operation"
+        },
+        "x-ms-examples": {
+          "Create or update an Active Directory connector instance.": {
+            "$ref": "./examples/CreateOrUpdateActiveDirectoryConnector.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "ActiveDirectoryConnectors"
+        ],
+        "operationId": "ActiveDirectoryConnectors_Delete",
+        "description": "Deletes an Active Directory connector resource",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionId"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/dataControllerName"
+          },
+          {
+            "$ref": "#/parameters/activeDirectoryConnectorName"
+          },
+          {
+            "$ref": "#/parameters/apiVersion"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
>"200" [](http://example.com/codeflow?start=10&length=5)

According to our RPC, you return either 200 or 204 upon deletion completion, not both: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#delete-resource-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/async-api-reference.md#delete-resource-asynchronously)

Do you have a situation where you can return both? Is the location header pointing to this resource?

Reply:
Not sure I follow, when there's no connection to Azure RPaaS sends a delete call and your RP returns 200 instead of 204?

Reply:
I'll sign off your PR since it seems like you are following the existing pattern from other APIs in this spec.

Reply:
I touched base with John. and let me further clarify... maybe I complicated the problem a little bit...

There are scenarios for deletion that is both synchronous and asynchronous. 

When it is not an Arc enabled scenario, deleting the resource does not need to talk with customer's cluster but merely remove metadata from Azure. In this case, it is a synchronous scenario, and 200 is returned.

When it is an Arc enabled scenario, deleting the resource needs to talk with customer's cluster and might take indefinite amount of time. In this case, it is an asynchronous scenario, and 202 is returned. 

Lastly, when the resource does not exist: 204 is returned.

I left John a note that this might need a bit of clarification in Rpaas-ARM scope and one partner engineer in the breaking change review board has the similar question.